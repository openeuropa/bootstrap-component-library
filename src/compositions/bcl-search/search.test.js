import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";
import demoData from "@openeuropa/bcl-search/dataSearch";

const template = "@oe-bcl/bcl-search/search.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - Search", () => {
  test(`renders correctly`, () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });
});
