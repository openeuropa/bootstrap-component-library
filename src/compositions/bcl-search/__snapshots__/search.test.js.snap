// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Search renders correctly 1`] = `
<jest>
  <main>
    <div
      class="container mt-5"
    >
      <div
        class="row"
      >
        <div
          class="col-md-3"
        >
          <div
            aria-labelledby="offcanvasExampleLabel"
            class="bcl-offcanvas offcanvas-start offcanvas"
            data-bs-scroll="true"
            id="bcl-offcanvas"
            tabindex="-1"
          >
            <div
              class="offcanvas-header p-lg-0"
            >
              <h4
                class="offcanvas-title mb-lg-4"
                id="offcanvasExampleLabel"
              >
                Filter options
              </h4>
              <button
                class="btn-close text-reset d-lg-none"
                data-bs-dismiss="offcanvas"
                type="button"
              />
            </div>
            <div
              class="offcanvas-body p-lg-0"
            >
              <form
                novalidate="true"
                onsubmit="return false;"
              >
                <div
                  class="mb-3"
                >
                  <input
                    class="form-control form-control-sm"
                    placeholder="Keyword"
                    type="text"
                  />
                </div>
                <div
                  class="mb-3"
                >
                  <label
                    class="form-label"
                  >
                    Content types
                  </label>
                  <select
                    class="form-select form-select-sm"
                  >
                    <option
                      value="1"
                    >
                      a select option
                    </option>
                    <option
                      value="2"
                    >
                      another select option
                    </option>
                    <option
                      value="3"
                    >
                      another option
                    </option>
                    <option
                      value="4"
                    >
                      last option
                    </option>
                    <option
                      selected=""
                      value=""
                    >
                      the selected option
                    </option>
                  </select>
                </div>
                <div
                  class="mb-3"
                >
                  <label
                    class="form-label"
                  >
                    Filter A
                  </label>
                  <select
                    class="multi-select"
                    multiple="true"
                  >
                    <option
                      value="1"
                    >
                      a select option
                    </option>
                    <option
                      value="2"
                    >
                      another select option
                    </option>
                    <option
                      value="3"
                    >
                      another option
                    </option>
                    <option
                      value="4"
                    >
                      last option
                    </option>
                    <option
                      selected=""
                      value=""
                    >
                      the selected option
                    </option>
                  </select>
                </div>
                <fieldset
                  class="mb-3"
                >
                  <legend
                    class="col-form-label"
                  >
                    Filter B
                  </legend>
                  <div
                    class="col-sm-10"
                  >
                    <div
                      class="form-check"
                    >
                      <input
                        checked="true"
                        class="form-check-input"
                        type="checkbox"
                      />
                      <label
                        class="form-check-label"
                      >
                        Checked checkbox
                      </label>
                    </div>
                    <div
                      class="form-check"
                    >
                      <input
                        class="form-check-input"
                        type="checkbox"
                      />
                      <label
                        class="form-check-label"
                      >
                        Default checkbox
                      </label>
                    </div>
                  </div>
                </fieldset>
                <fieldset
                  class="mb-3"
                >
                  <legend
                    class="col-form-label"
                  >
                    Filter C
                  </legend>
                  <input
                    class="mb-2 form-control form-control-sm"
                    type="date"
                  />
                  <input
                    class="form-control form-control-sm"
                    type="date"
                  />
                </fieldset>
                <div
                  class="mt-4"
                >
                  <button
                    class="me-2 btn btn-primary btn-md"
                    type="submit"
                  >
                    Refine
                  </button>
                  <button
                    class="btn btn-secondary btn-md"
                    type="submit"
                  >
                    Clear
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
        <div
          class="col-12 col-lg-9"
        >
          <button
            aria-controls="bcl-offcanvas"
            class="d-lg-none w-100 mb-4 btn btn-primary btn-md"
            data-bs-target="#bcl-offcanvas"
            data-bs-toggle="offcanvas"
            type="button"
          >
            Filter options
            <svg
              class="ms-2 bi icon--xs"
            >
              <use
                xlink:href="#sliders"
              />
            </svg>
          </button>
          <div
            class="mb-3 clearfix"
          >
            <div
              class="float-end d-none d-md-block"
            >
              <label
                class="form-label"
              >
                Sort by
              </label>
              <select
                class="mb-4 form-select form-select-sm"
              >
                <option
                  value="1"
                >
                  a select option
                </option>
                <option
                  value="2"
                >
                  another select option
                </option>
                <option
                  value="3"
                >
                  another option
                </option>
                <option
                  value="4"
                >
                  last option
                </option>
                <option
                  selected=""
                  value=""
                >
                  A-Z
                </option>
              </select>
            </div>
            <h4
              class="mb-4"
            >
              Search results (18)
            </h4>
            <span
              class="me-2 badge bg-primary"
            >
              News
              <span
                aria-hidden="true"
                class="icon--close"
              >
                <svg
                  class="bi icon--s"
                >
                  <use
                    xlink:href="#x"
                  />
                </svg>
              </span>
            </span>
            <span
              class="me-2 badge bg-primary"
            >
              Item 01
              <span
                aria-hidden="true"
                class="icon--close"
              >
                <svg
                  class="bi icon--s"
                >
                  <use
                    xlink:href="#x"
                  />
                </svg>
              </span>
            </span>
            <span
              class="me-2 badge bg-primary"
            >
              Item 02
              <span
                aria-hidden="true"
                class="icon--close"
              >
                <svg
                  class="bi icon--s"
                >
                  <use
                    xlink:href="#x"
                  />
                </svg>
              </span>
            </span>
          </div>
          <hr
            class="d-none d-md-block mb-4"
          />
          <div
            class="d-md-none"
          >
            <label
              class="form-label"
            >
              Sort by
            </label>
            <select
              class="mb-4 form-select form-select-sm"
            >
              <option
                value="1"
              >
                a select option
              </option>
              <option
                value="2"
              >
                another select option
              </option>
              <option
                value="3"
              >
                another option
              </option>
              <option
                value="4"
              >
                last option
              </option>
              <option
                selected=""
                value=""
              >
                A-Z
              </option>
            </select>
          </div>
          <div
            class="border-0 mb-5 card"
          >
            <div
              class="row g-0"
            >
              <div
                class="col-md-3"
              >
                <img
                  alt="alt img"
                  class="card-img-top"
                  src="https://picsum.photos/200/200?random=1"
                />
              </div>
              <div
                class="col-md-9"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title mb-4"
                  >
                    Lorem ipsum dolor sit amet
                  </h5>
                  <p
                    class="card-text mb-4"
                  >
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
                  </p>
                  <span
                    class="badge bg-primary"
                  >
                    Item 01
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="border-0 mb-5 card"
          >
            <div
              class="row g-0"
            >
              <div
                class="col-md-3"
              >
                <img
                  alt="alt img"
                  class="card-img-top"
                  src="https://picsum.photos/200/200?random=2"
                />
              </div>
              <div
                class="col-md-9"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title mb-4"
                  >
                    Incididunt ut labore et dolore
                  </h5>
                  <p
                    class="card-text mb-4"
                  >
                    Amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore adipiscing elit, sed do eiusmod tempor incididunt ut labore.
                  </p>
                  <span
                    class="badge bg-primary me-2"
                  >
                    Item 01
                  </span>
                  <span
                    class="badge bg-primary"
                  >
                    Item 02
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="border-0 mb-5 card"
          >
            <div
              class="row g-0"
            >
              <div
                class="col-md-3"
              >
                <img
                  alt="alt img"
                  class="card-img-top"
                  src="https://picsum.photos/200/200?random=3"
                />
              </div>
              <div
                class="col-md-9"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title mb-4"
                  >
                    Quis nostrud exercitation ullamco
                  </h5>
                  <p
                    class="card-text mb-4"
                  >
                    Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                  </p>
                  <span
                    class="badge bg-primary me-2"
                  >
                    Item 01
                  </span>
                  <span
                    class="badge bg-primary"
                  >
                    Item 02
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="border-0 mb-5 card"
          >
            <div
              class="row g-0"
            >
              <div
                class="col-md-3"
              >
                <img
                  alt="alt img"
                  class="card-img-top"
                  src="https://picsum.photos/200/200?random=4"
                />
              </div>
              <div
                class="col-md-9"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title mb-4"
                  >
                    Quis nostrud exercitation ullamco
                  </h5>
                  <p
                    class="card-text mb-4"
                  >
                    Quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                  </p>
                  <span
                    class="badge bg-primary"
                  >
                    Item 01
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="border-0 mb-5 card"
          >
            <div
              class="row g-0"
            >
              <div
                class="col-md-3"
              >
                <img
                  alt="alt img"
                  class="card-img-top"
                  src="https://picsum.photos/200/200?random=5"
                />
              </div>
              <div
                class="col-md-9"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title mb-4"
                  >
                    Duis aute irure dolor in Excepteur
                  </h5>
                  <p
                    class="card-text mb-4"
                  >
                    Duis aute irure dolor in Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                  </p>
                  <span
                    class="badge bg-primary me-2"
                  >
                    Item 01
                  </span>
                  <span
                    class="badge bg-primary"
                  >
                    Item 02
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="border-0 mb-5 card"
          >
            <div
              class="row g-0"
            >
              <div
                class="col-md-3"
              >
                <img
                  alt="alt img"
                  class="card-img-top"
                  src="https://picsum.photos/200/200?random=6"
                />
              </div>
              <div
                class="col-md-9"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title mb-4"
                  >
                    Excepteur sint occaecat cupidatat non
                  </h5>
                  <p
                    class="card-text mb-4"
                  >
                    Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                  </p>
                  <span
                    class="badge bg-primary"
                  >
                    Item 02
                  </span>
                </div>
              </div>
            </div>
          </div>
          <hr />
          <nav
            class="mt-5"
          >
            <ul
              class="pagination justify-content-center"
            >
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href="#"
                >
                  Previous
                </a>
              </li>
              <li
                class="page-item disabled"
              >
                <a
                  aria-disabled="true"
                  class="page-link"
                  href="#"
                  tabindex="-1"
                >
                  1
                </a>
              </li>
              <li
                class="page-item active"
              >
                <a
                  aria-current="page"
                  class="page-link"
                  href="#"
                >
                  2
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href="#"
                >
                  3
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href="#"
                >
                  Next
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </main>
</jest>
`;
